django_ontruck/monads/result.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import annotations
+
+from abc import ABCMeta, abstractmethod
+from typing import Callable, Optional, TypeVar, Generic, Union
+
+
+ValueType = TypeVar('ValueType')
+U = TypeVar('U')  # generic type for values
+
+ErrorType = TypeVar('ErrorType')
+F = TypeVar('F')  # generic type for errors
+
+
+class Result(Generic[ValueType, ErrorType], metaclass=ABCMeta):
+    """Result monad interface.
+    """
+    _inner_value: Union[ValueType, ErrorType]
+
+    @abstractmethod
+    def is_ok(self) -> bool:
+        """Returns True if `self` is an `Ok`
+        """
+        return NotImplemented
```

Comment:
You want to raise `NotImplementedError` here I think.  `NotImplemented` is used to indicate that you are delegating the handling of a particular method to another method.  (For example when you want to use `__radd__` instead of `__add__`.)

https://jcalderone.livejournal.com/32837.html